src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -158692,5 +158692,45 @@ class Program
                 //         y.Value.Item.ToString();
                 Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "y.Value.Item").WithLocation(15, 9));
         }
+
+        [Fact, WorkItem("https://devdiv.visualstudio.com/DevDiv/_workitems/edit/2045970")]
+        public void BadAsyncLambdaInNamedArgument()
+        {
+            var source = """
+                #nullable enable
+                using System;
+                using System.Threading.Tasks;
+
+                class D
+                {
+                    void M()
+                    {
+                        var c1 = new C();
+                        var c2 = new C();
+                        C.M1(f: async () =>
+                        {
+                            if (c2 != null)
+                            {
+                                c1.
+                                await c2.M2();
+                            }
+                        });
+                    }
+                }
+
+                class C
+                {
+                    public static void M1(Func<Task> f) { }
+                    public async Task M2() => await Task.Yield();
+                }
+                """;
+            CreateCompilation(source).VerifyDiagnostics(
```

Comment:
nit: consider adjusting the scenario so that a nullable diagnostic is expected or something like that, to show that nullable analysis is definitely occurring here.

Reply:
If you mean inside the lambda, no diagnostics will be reported there since it's a bad call and hence diagnostics are disabled inside:

https://github.com/dotnet/roslyn/blob/c23414e1365f21c6bd34ab5b007546066f25b5d4/src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs#L6743-L6747